### PR TITLE
Feat/Login state caching

### PR DESCRIPTION
**Pull request by SaturneV**
### Summary

This Pull Request implements login caching, meaning that once the user is logged in with his google account, the application starts directly on the Map. This feature also implements the ability to see the user score and username on the application even when not connected to the Internet.

Implementing this function required firstly implementing a function that checks the application's connectivity to decide whether or not to load the user from the database or cache. It was also necessary to implement a simple Splash Screen to wait for the user to load before displaying the application.

This PR cherry picks commits from the soon to be closed PR https://github.com/SwEnt-Group8/Street-work-app/pull/166 due to features separation.

### Changes

- Add `NewtorkUtils.kt` file.
- Implement `isInternetAvailable` function in `NewtorkUtils.kt` file.
- Add a `SplashScreen.kt` file with a `SplashScreen` composable.
- Implement the login caching logic in `MainActivity.kt`.
- Implement the login caching logic in `SignIn.kt`.

### Tests

- Modify tests to implement the new feature signature and parameters.
- Add an UI test to test the `SplashScreen` composable.
- Add a `StreetWorkAppMainTest` file with tests.

**Please fully test this feature with your device / emulator to make sure it works properly.**

### CODE REVIEW

**Comment by tercierp** [comment]
After some tests there is a problem with the markers when you get connected for the second time.
It's as if calling up the map doesn't call up the markers, but I'm connected and if I change screen and come back to the map the markers appear.
<p align="center">
  image
</p>

**Review by tercierp — CHANGES_REQUESTED** [review]
I'll let you respond to my comments, especially the one about markers disappearing after a user cache in. Thanks for your contribution!

File: app/src/main/java/com/android/streetworkapp/MainActivity.kt [hunk 1]
```diff
@@ -120,16 +138,60 @@ fun StreetWorkAppMain(testInvokation: NavigationActions.() -> Unit = {}) {
...
+  }
+
+  // Display splash screen while determining start destination and preferences
+  if (resolvedStartDestination == null || !resolvedPreferencesParameters) {
+    SplashScreen()
+  } else {
+    if (loginState == true) {
+      if (internetAvailable) {
+        Log.d("MainActivity", "Internet available, fetching user $uid from database")
+        userViewModel.getUserByUidAndSetAsCurrentUser(uid!!)
```
[the remark this answers is not in the document]
Comment (SaturneV) [reply app/src/main/java/com/android/streetworkapp/MainActivity.kt]: I agree with you, I added a check.

**Comment by SaturneV** [comment]
I fixed the map that wasn't displaying the markers by forcing the map to load the parks when starting the composable. I'll let you check if it works on your side. Good catch!

**Review by tercierp — APPROVED** [review]

**Comment by tercierp** [comment]
Good patch this time I can't break the markers on the map, merged !

**Pull request merged**